xds/internal/clients/xdsclient/authority.go

```diff
@@ -720,6 +731,10 @@ func (a *authority) unwatchResource(rType ResourceType, resourceName string, wat
 			// there when the watch was registered.
 			resources := a.resources[rType]
 			state := resources[resourceName]
+			if state == nil {
+				a.logger.Warningf("Attempting to unwatch resource %q of type %q which is not currently watched", resourceName, rType.TypeName)
+				return
+			}
```

Comment:
Is this condition possible? `unwatchResource` is returned as the cancel function from `watchResource` only when a watch is registered.